### PR TITLE
refactor!: remove deprecated GeneratedVaadinCheckboxGroup

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

1. Removed deprecated `GeneratedVaadinCheckboxGroup` class,
2. Updated `CheckboxGroup` to implement `HasThemeVariant`.

## Type of change

- Breaking change

### CODE REVIEW

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java [hunk 1]
```diff
@@ -26,19 +26,27 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import com.vaadin.flow.component.AbstractSinglePropertyField;
 import com.vaadin.flow.component.AttachEvent;
 import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
```
Comment (tomivirkki) [inline vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java:33]: nit: unused imports

**Review by tomivirkki — APPROVED** [review]

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java [hunk 2]
```diff
@@ -26,19 +26,27 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import com.vaadin.flow.component.AbstractSinglePropertyField;
 import com.vaadin.flow.component.AttachEvent;
 import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEvent;
```
Comment (web-padawan) [inline vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java:32]:
```suggestion
```

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java (hunk 1 above)
Comment (web-padawan) [inline vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java:33]:
```suggestion
```

Comment (web-padawan) [reply vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java]: Removed.

**Pull request merged**